jmsSource/src/main/java/io/vantiq/extsrc/jmsSource/JMSCore.java

```diff
@@ -64,25 +64,16 @@
         public void handleMessage(ExtensionServiceMessage message) {
             log.trace("Reconnect message received. Reinitializing configuration");
 
+            // Do connector-specific stuff here
             jmsConfigHandler.configComplete = false;
 
-            CompletableFuture<Boolean> success = client.connectToSource();
-
-            try {
-                if ( !success.get(CONNECTION_TIMEOUT, TimeUnit.SECONDS) ) {
-                    if (!client.isOpen()) {
-                        log.error("Failed to connect to server url '" + targetVantiqServer + "'.");
-                    } else if (!client.isAuthed()) {
-                        log.error("Failed to authenticate within 10 seconds using the given authentication data.");
-                    } else {
-                        log.error("Failed to connect within 10 seconds");
-                    }
+            // Boiler-plate reconnect method, if reconnect fails then we call close()
+            CompletableFuture<Boolean> reconnectResult = client.doCoreReconnect();
+            reconnectResult.thenAccept(success -> {
```

Comment:
It may be worth noting here (boilerplate comment) that `reconnectResult` can't complete until the handler completes.  When I read the code, I thought maybe `thenAccept` may be in a race condition with the async call above it.  I don't really think that's the case anyway, but, in fact, it can't happen anyway (due to the but we're fixing).